atox/src/main/kotlin/ToxService.kt

```diff
@@ -124,6 +132,24 @@ class ToxService : LifecycleService() {
                 }
         }
 
+        friendRequestManager.getAll()
+            .filterNotNull()
+            .asLiveData()
+            .observe(this@ToxService) { friendRequests ->
+                if (knownFriendRequests.isEmpty() && friendRequests.isEmpty()) {
```

Comment:
If these are both empty, none of the lines after will do anything, so you can remove this special case.

Reply:
I thought you were going to say that. So, I tested it thoroughly and it actually gets triggered on boot.

Reply:
Fair enough that we enter this if-case and return, but removing this should be a no-op since nothing after this line will have any effect if both containers are empty? :P

Reply:
Yes, but return early. :D